src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqExpectationsRepository.kt

```diff
@@ -87,6 +85,40 @@ class JooqExpectationsRepository(dsl: DSLContext)
                 }
     }
 
+    override fun getAllExpectations(): List<TouchstoneModelExpectations>
+    {
+        val records = dsl.select(
+                SCENARIO_DESCRIPTION.DISEASE,
+                TOUCHSTONE.ID,
+                RESPONSIBILITY_SET.MODELLING_GROUP,
+                *BURDEN_ESTIMATE_EXPECTATION.fields(),
+                Tables.outcomes.OUTCOME)
+                .fromJoinPath(RESPONSIBILITY, BURDEN_ESTIMATE_EXPECTATION)
+                .joinPath(RESPONSIBILITY, SCENARIO, SCENARIO_DESCRIPTION)
+                .joinPath(RESPONSIBILITY, RESPONSIBILITY_SET, TOUCHSTONE)
+                .joinPath(BURDEN_ESTIMATE_EXPECTATION, Tables.outcomes, joinType = JoinType.LEFT_OUTER_JOIN)
+                .where(RESPONSIBILITY.IS_OPEN.eq(true))
+                .and(TOUCHSTONE.STATUS.eq("open"))
+                .and(SCENARIO.TOUCHSTONE.eq(RESPONSIBILITY_SET.TOUCHSTONE))
+                .orderBy(RESPONSIBILITY_SET.MODELLING_GROUP, SCENARIO_DESCRIPTION.DISEASE, TOUCHSTONE.ID)
+
+        val outcomes = records.groupingBy{ it[BURDEN_ESTIMATE_EXPECTATION.ID] }
```

Comment:
So this quite ugly bit of code was an attempt to get all the outcomes along with the estimates stuff, and save doing separate queries to get each of the outcomes sets (what withOutcomesAndCountries would do, I think). Not sure if there's a nicer approach?

Reply:
Wow, that is quite something, you've definitely discovered new Kotlin syntax that I've never seen before!
But I think this is clearer:

```
        val outcomes = records.groupBy { it[BURDEN_ESTIMATE_EXPECTATION.ID] }
                .mapValues {
                    it.value.mapNotNull { row -> row[Tables.outcomes.OUTCOME] }
                }
```